src/map/ai/states/item_state.cpp

```diff
@@ -278,3 +279,20 @@ bool CItemState::HasMoved()
     return floorf(m_startPos.x * 10 + 0.5f) / 10 != floorf(m_PEntity->loc.p.x * 10 + 0.5f) / 10 ||
            floorf(m_startPos.z * 10 + 0.5f) / 10 != floorf(m_PEntity->loc.p.z * 10 + 0.5f) / 10;
 }
+
+// Special case for Soultrapper usage:
+// Valid to use on mobs that are: unclaimed, claimed by you, claimed by someone else
+// This is handled this way to avoid bringing in a new very specialized targetting flag
+// just for soultrapping.
+CBattleEntity* CItemState::HandleSoultrapperUse()
+{
+    if (m_PItem->isSoultrapper())
+    {
+        // Reset possible "already claimed" error from previous lookup
+        m_errorMsg.reset();
+
+        // Call CBattleEntity's simpler IsValidTarget()
+        CState::UpdateTarget(m_PEntity->CBattleEntity::IsValidTarget(m_targid, m_PItem->getValidTarget(), m_errorMsg));
+    }
+    return static_cast<CBattleEntity*>(CState::GetTarget());
+}
```

Comment:
I hate this. Like, a lot. There _has_ to be a better way to handle this that _isn't_:
- Adding exceptions into `CItemState` just for Soultrapping
- Adding a `validTargets` filter JUST for mobs that are: unclaimed, claimed by you, claimed by someone else

Anyone got any ideas? I'm drawing a blank on this one

Reply:
Is there another flag we could add, that isn't as broad as `TARGET_MOB`?
https://github.com/LandSandBoat/server/blob/094ec9f022916bc86b8d0efee166b626ec560c2c/src/map/entities/battleentity.h

Reply:
`TARGET_CLAIM` because `TARGET_ENEMY` should already cover mobs _except_ when they are claimed by you. So we just skip past conditions of caring about the claim status if we see that flag as well. Or keep it a hard coded item check for now, since we don't yet know of any other items we'd need it for.

Reply:
Refactored this to require slightly less eye-bleach